code/modules/projectiles/guns/ballistic/automatic.dm

```diff
@@ -352,3 +352,192 @@
+		from so many calculation errors.")
+
+/obj/item/gun/ballistic/automatic/battle_rifle/examine(mob/user)
+	. = ..()
+	if(shots_before_degradation)
+		. += span_notice("\The [src] can fire [shots_before_degradation] more times before risking system degradation.")
+	else
+		. += span_notice("\The [src] is in the process of system degradation. It is currently at stage [degradation_stage] of [degradation_stage_max]. Use a multitool on [src] to recalibrate. Alternatively, insert it into a weapon recharger.")
+
+/obj/item/gun/ballistic/automatic/battle_rifle/update_icon_state()
+	. = ..()
+	if(!shots_before_degradation)
+		inhand_icon_state = "[base_icon_state]-empty"
+	else
+		inhand_icon_state = "[base_icon_state]"
+	return ..()
+
+/obj/item/gun/ballistic/automatic/battle_rifle/update_overlays()
+	. = ..()
+	if(degradation_stage)
+		. += "[base_icon_state]_empty"
+	else if(shots_before_degradation)
+		var/ratio_for_overlay = CEILING(clamp(shots_before_degradation / max_shots_before_degradation, 0, 1) * 3, 1)
+		. += "[icon_state]_stage_[ratio_for_overlay]"
+
+/obj/item/gun/ballistic/automatic/battle_rifle/emp_act(severity)
+	. = ..()
+	if (!(. & EMP_PROTECT_SELF) && prob(50 / severity))
+		shots_before_degradation = 0
+		emp_malfunction = TRUE
+		attempt_degradation(TRUE)
+
+/obj/item/gun/ballistic/automatic/battle_rifle/emag_act(mob/user, obj/item/card/emag/emag_card)
+	. = ..()
+	if(obj_flags & EMAGGED)
+		return FALSE
+	obj_flags |= EMAGGED
+	projectile_damage_multiplier = emagged_projectile_damage_multiplier
+	balloon_alert(user, "heat distribution systems deactivated")
+	return TRUE
+
+/obj/item/gun/ballistic/automatic/battle_rifle/multitool_act(mob/living/user, obj/item/tool)
+	. = ..()
+	if(!tool.use_tool(src, user, 20 SECONDS, volume = 50))
+		balloon_alert(user, "interrupted!")
+		return
+
+	if(emp_malfunction)
+		emp_malfunction = FALSE
+
+	shots_before_degradation = initial(shots_before_degradation)
+	degradation_stage = initial(degradation_stage)
+	projectile_speed_multiplier = initial(projectile_speed_multiplier)
+	fire_delay = initial(fire_delay)
+	update_appearance()
+	balloon_alert(user, "system reset.")
+
+/obj/item/gun/ballistic/automatic/battle_rifle/try_fire_gun(atom/target, mob/living/user, params)
+	. = ..()
+	if(!chambered || chambered && !chambered.loaded_projectile)
+		return
+
+	if(shots_before_degradation)
+		shots_before_degradation --
+		return
+
+	else if (obj_flags & EMAGGED && degradation_stage == degradation_stage_max && !explosion_timer)
+		perform_extreme_malfunction(user)
+
+	else
+		attempt_degradation(FALSE)
+
+
+/obj/item/gun/ballistic/automatic/battle_rifle/process_fire(atom/target, mob/living/user, message = TRUE, params = null, zone_override = "", bonus_spread = 0)
+	if(chambered.loaded_projectile && emp_malfunction && prob(75) || chambered.loaded_projectile && degradation_stage == degradation_stage_max && prob(75))
```

Comment:
Same here with the parenthesis
```suggestion
	if((chambered.loaded_projectile && emp_malfunction && prob(75)) || (chambered.loaded_projectile && degradation_stage == degradation_stage_max && prob(75)))
```
But also it's a bit more complex than it needs to be
```suggestion
	if(chambered.loaded_projectile && prob(75) && (emp_malfunction || degradation_stage == degradation_stage_max))
```